CHANGELOG.md

```diff
@@ -1,5 +1,12 @@
 ## vNext (TBD)
 
+### Deprecations
+* The ECMAScript Array APIs (such as `map`, `every`, access using indexing operator `[]`, etc.) on the `BaseSubscriptionSet` type was deprecated.
```

Comment:
```suggestion
* The ECMAScript Array APIs (such as `map`, `every`, access using indexing operator `[]`, etc.) on the `SubscriptionSet` types was deprecated.
```

Ditto below.

Personally, I think it would be best if we treated `BaseSubscriptionSet` as an implementation detail for how we remove redundancy in our type declarations rather than something user facing. Feel free to ignore if the rest of the team disagrees.

Reply:
I think it makes sense changing it to `SubscriptionSet` publicly 👍